### PR TITLE
Register components

### PR DESCRIPTION
**Pull request by rcsharp5**
**https://chartiq.kanbanize.com/ctrl_board/18/cards/10931/details**

**Description of change**
-This makes the catalog work and includes config driven apps

**Description of testing**
- The whole catalog/launcher

### CODE REVIEW

File: src-built-in/components/myApps/src/modules/FDC3.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		});
+	}
+
+	/**
+     * 
+     * @param {string} path The restful method path
+     * @param {object} params The post data
+     * @param {function} done The callback function
+     */
+	_post(path, params, done) {
```
Comment (carterbs) [inline src-built-in/components/myApps/src/modules/FDC3.js]: Is this whole file a duplicate?

File: src-built-in/components/myApps/src/stores/StoreActions.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import { getStore } from './LauncherStore'
-
+import {getStore} from "./LauncherStore";
+import AppDirectory from "../modules/AppDirectory";
```
Comment (carterbs) [inline src-built-in/components/myApps/src/stores/StoreActions.js]: same comments as the other file that looks exactly like this. The data thing is just weird.

**Comment by carterbs** [comment]
Comments
------
* The app catalog scrolls vertically. After speaking with nicolai this is intentional so that if there are 4 or 5 carousels, the window will scroll. WIth 2 carousels it feels like a bug.
* The tags button is flush to the top right of the window. Feels weird.
* Navigate to an app. Click a tag at the bottom. Remove the tag from the filter. The app goes back to "home". It feels like it should go back to the app's details?

Issues
------
* Window title shouldn't be select-able.
* Nothing happens when clicking an email link. Either make it not a link or make it do something.
* I get the "You don't have permission to add this app" banner at the top for a component that I've already added. This shouldn't be possible.
* After clicking into an app's details, the search at the top is inoperable.
* Welcome component is blank window.
* Spawn a welcome component. Resize it in the direction of the app catalog. App catalog resizes. It shouldn't.
* Launcher -- Add notepad, app catalog as favorites. Try to add welcome component. Doesn't go to the toolbar.

File: src-built-in/components/myApps/src/components/AddNewAppForm.jsx [hunk 3]
```diff
@@ -14,78 +14,78 @@ import {
...
+	}
+	/**
      * Clears all form inputs
      */
-    onClear() {
-        document.getElementById("new-app").reset()
-    }
-    /**
+	onClear() {
+		document.getElementById("new-app").reset();
```
[the remark this answers is not in the document]
Comment (amaroks) [reply src-built-in/components/myApps/src/components/AddNewAppForm.jsx]: I know using ref might be a better idea, but thought its just a safe, simple form reset.

File: src-built-in/components/myApps/src/components/AppDefinition.jsx [hunk 4]
```diff
@@ -26,51 +26,39 @@ export default class AppDefinition extends React.Component {
...
-			}, () => {
-				pendingSpawn = false
-			})
-			return
-		}
-		// Otherwise launch application by name
-		FSBL.Clients.LauncherClient.spawn(name, {}, (err, data) => {
-			pendingSpawn = false
+		FSBL.Clients.LauncherClient.spawn(this.props.app.name, null, (err, data) => {
+			console.log("pending spawn complete");
```
[the remark this answers is not in the document]
Comment (amaroks) [reply src-built-in/components/myApps/src/components/AppDefinition.jsx]: What is lawg?

[the remark this answers is not in the document]
Comment (carterbs) [reply src-built-in/components/myApps/src/components/AppDefinition.jsx]: log. but i was tired. so it's "log", said in a southern american accent.

File: src-built-in/components/appLauncher/src/stores/appLauncherStore.js [hunk 5]
```diff
@@ -67,6 +67,7 @@ var Actions = {
 	},
 	//get adhoc components that the user created from storage.
 	getUserDefinedComponentList(cb) {
+		return cb();
```
[the remark this answers is not in the document]
Comment (carterbs) [reply src-built-in/components/appLauncher/src/stores/appLauncherStore.js]: Better comment: Why are we returning and doing nothing. is this intentional or an accident? Needs a comment either way.

File: src-built-in/components/myApps/src/components/AddNewAppForm.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (carterbs) [reply src-built-in/components/myApps/src/components/AddNewAppForm.jsx]: I'll let @clausjs weigh in on best practices here. If refs are the way to go, we should use them.

File: src-built-in/components/appCatalog/src/components/Showcase/Header.jsx [hunk 6]
```diff
@@ -29,6 +48,10 @@ const Header = props => {
 		storeActions.removeApp(props.appId);
 	}
 
+	const launchApp = () => {
```
[the remark this answers is not in the document]
Comment (carterbs) [reply src-built-in/components/appCatalog/src/components/Showcase/Header.jsx]: @clausjs , I apologize. This question mark is a bad github comment. I should have said "Why is a function that does nothing at all in the code?"

Also, I'd prefer components, stateless components included, to be in their own files. Makes things nice and simple.

**Pull request merged**